src/shared/clients/lambdaClient.ts

```diff
@@ -96,7 +115,10 @@ export class DefaultLambdaClient {
         }
     }
 
-    private async createSdkClient(): Promise<Lambda> {
-        return await globals.sdkClientBuilder.createAwsService(Lambda, undefined, this.regionCode)
+    private async getSdkClient(): Promise<Lambda> {
+        if (!this.client) {
+            this.client = await globals.sdkClientBuilder.createAwsService(Lambda, undefined, this.regionCode)
+        }
+        return this.client
```

Comment:
is this for performance? I would skip this for now. The SDKs cached some things so it's not always clear when a new client is needed or not. 

Eventually, we will solve this with an abstraction that generalizes this. Meanwhile, it may avoid some pain if we use the same pattern in all of the `fooClient.ts` modules.

Reply:
This change was done due to a previous PR comment recommending to not stub the `getSdkClient()` method since it was private. The goal was to allow me to pass in a mock client for testing, otherwise it created its own. 
But now I see this has a side effect due to the SDK caching you mentioned.

Do you recommend I revert this and stub the private method, or do you see another way to do this? I need to stub the client either way since it makes a network call.

Reply:
I didn't realize we needed to worry about cache clearance. I've removed this change and instead provide the option to provide a client to the method itself, this is so I can still test without needing to make an actual network call.